neptune/new/internal/operation_processors/async_operation_processor.py

```diff
@@ -84,6 +84,8 @@ def enqueue_operation(self, op: Operation, wait: bool) -> None:
             self.wait()
 
     def wait(self):
+        if not self._consumer.is_running():
+            raise Exception("Synchronization job is not running, unable to synchronize")
```

Comment:
@normandy7 Some nice error message could be useful here.

The case is:
1. User starts a Run/Model/Version in ASYNC mode.
2. Asynchronous syncing thread dies from any reason (limits reached/connection errors/bugs/whatever).
3. Some error message with details about what caused async thread to die is presented printed.
4. User's script continue processing in "offline" mode with all the data being stored on local disk.
5. User calls `wait()` or `sync()`, but synchronization cannot be finished since syncing thread has been already dead for some time.

Reply:
Are these on the right track?

* If it happens in the middle of script execution, i.e. `wait()`/`sync()`: `The waiting/syncing was interrupted because the data synchronization process had already stopped.`
* If it happens on stop: `The synchronization process had already stopped before Neptune could finish the remaining operations.`

Are more solution suggestions or context needed?

Reply:
Yeah. They are on right track. I don't know if we use term "offline mode" anywhere in this context but it's kinda similar to being switched from async to offline mode. In real offline mode we continue without any warning. @Herudaio should we throw an exception in this case or just print a warning?

Reply:
Disclaimer: I'm not sure I have the full context here.

If it behaves similar to offline mode, I would follow behavior from offline mode, so if I understand correctly warning would be more fitting.

Reply:
Full context is in first comment and it would be better if you feel you have it. ;)
In our engineering side of view it's similar to offline mode but I have no idea about user's side of view and how you usually communicate this situation to user. I don't want to introduce any inconsistencies in communication and I don't have full context here as well. :)

Reply:
Re-read this with more coffee. We do describe it in docs as "switching to offline mode", so a warning would be good, but "as in offline mode" would be a guiding principle.

Reply:
1. Something like "cannot finish synchronization data with server due to synchronization thread being dead" at the beginning would be nice.
2. Any suggestions for name of the exception? How about `NeptuneSynchronizationThreadDead`?

Reply:
@normandy7 @Herudaio What do you think?

Reply:
I'm not sure the user cares about the inner workings of the thread, specifically :) What's relevant to them is that the synchronization stopped at some point, but the logging continued offline. So they need to understand that they should look into two things:

1. They need to sync the offline data manually
1. Is there some reason (within their control) that the synching was interrupted?

If the above assumptions are correct, and one error is enough, how about something like this:

```
The synchronization process had stopped before Neptune could finish uploading the logged metadata.
Your data is stored locally, but you'll need to finish the synchronization manually.
To synchronize with the Neptune servers, enter the following on your command line:

    neptune sync

For details, see https://docs.neptune.ai/you-should-know/connection-modes#uploading-offline-data

If the synchronization fails, you may want to check your connection and ensure that you're
within limits by going to your Neptune project settings -> Usage.
If the issue persists, our support is happy to help.
```

Reply:
As for the exception name, I'd maybe rephrase the part that involves threads dying. Maybe `NeptuneSynchronizationAlreadyStopped`?